components/match2/wikis/rainbowsix/match_group_input_custom.lua

```diff
@@ -39,186 +33,133 @@ local CustomMatchGroupInput = {}
 
-	return match
-end
+	local opponents = Array.mapIndexes(function(opponentIndex)
+		return MatchGroupInput.readOpponent(match, opponentIndex, {})
+	end)
+	local games = MatchFunctions.extractMaps(match, #opponents)
+	match.bestof = MatchGroupInput.getBestOf(nil, games)
+	games = MatchFunctions.removeUnsetMaps(games)
+
+	local autoScoreFunction = MatchGroupInput.canUseAutoScore(match, opponents)
+		and MatchFunctions.calculateMatchScore(games, match.bestof)
+		or nil
+	Array.forEach(opponents, function(opponent, opponentIndex)
+		opponent.score, opponent.status = MatchGroupInput.computeOpponentScore({
+			walkover = match.walkover,
+			winner = match.winner,
+			opponentIndex = opponentIndex,
+			score = opponent.score,
+		}, autoScoreFunction)
+	end)
 
--- called from Module:Match/Subobjects
----@param map table
----@return table
-function CustomMatchGroupInput.processMap(map)
-	map = mapFunctions.getExtraData(map)
-	map = mapFunctions.getScoresAndWinner(map)
+	match.finished = MatchGroupInput.matchIsFinished(match, opponents)
 
-	return map
-end
+	if match.finished then
+		match.resulttype = MatchGroupInput.getResultType(winnerInput, finishedInput, opponents)
+		match.walkover = MatchGroupInput.getWalkover(match.resulttype, opponents)
+		match.winner = MatchGroupInput.getWinner(match.resulttype, winnerInput, opponents)
+		MatchGroupInput.setPlacement(opponents, match.winner, 1, 2)
+	end
 
----@param record table
----@param timestamp integer
-function CustomMatchGroupInput.processOpponent(record, timestamp)
-	local opponent = Opponent.readOpponentArgs(record)
-		or Opponent.blank()
+	MatchFunctions.getTournamentVars(match)
 
-	-- Convert byes to literals
-	if Opponent.isBye(opponent) then
-		opponent = {type = Opponent.literal, name = 'BYE'}
-	end
+	match.stream = Streams.processStreams(match)
+	match.links = MatchFunctions.getLinks(match)
 
-	---@type number|string
-	local teamTemplateDate = timestamp
-	-- If date is default date, resolve using tournament dates instead
-	-- default date indicates that the match is missing a date
-	-- In order to get correct child team template, we will use an approximately date and not the default date
-	if teamTemplateDate == DateExt.defaultTimestamp then
-		teamTemplateDate = Variables.varDefaultMulti('tournament_enddate', 'tournament_startdate', NOW)
-	end
+	match.extradata = MatchFunctions.getExtraData(match)
 
-	Opponent.resolve(opponent, teamTemplateDate)
-	MatchGroupInput.mergeRecordWithOpponent(record, opponent)
-end
+	match.games = games
+	match.opponents = opponents
 
---- Fetch information about the tournament
----@param data table
----@return table
-function CustomMatchGroupInput.getTournamentVars(data)
-	data.mode = Logic.emptyOr(data.mode, Variables.varDefault('tournament_mode', 'team'))
-	return MatchGroupInput.getCommonTournamentVars(data)
+	return match
 end
 
----@param data table
----@param indexedScores table[]
----@return table
+CustomMatchGroupInput.processMap = FnUtil.identity
```

Comment:
how many wikis set this by now?
like imo we should make this the default eventually...

Reply:
After the refactor it will likely be 0